crates/wasmtime/src/runtime/vm/component.rs

```diff
@@ -148,6 +184,37 @@ pub struct VMComponentContext {
     _marker: marker::PhantomPinned,
 }
 
+/// Represents the state of a stream or future handle.
+#[derive(Debug, Eq, PartialEq)]
+pub enum StreamFutureState {
+    /// Both the read and write ends are owned by the same component instance.
+    Local,
+    /// Only the write end is owned by this component instance.
+    Write,
+    /// Only the read end is owned by this component instance.
+    Read,
+    /// A read or write is in progress.
+    Busy,
+}
+
+/// Represents the state of a waitable handle.
+#[derive(Debug)]
+pub enum WaitableState {
+    /// Represents a task handle.
+    Task,
+    /// Represents a stream handle.
+    Stream(TypeStreamTableIndex, StreamFutureState),
+    /// Represents a future handle.
+    Future(TypeFutureTableIndex, StreamFutureState),
+}
+
+/// Represents the state associated with an error context
+#[derive(Debug, PartialEq, Eq, PartialOrd)]
+pub struct ErrorContextState {
+    /// Debug message associated with the error context
+    pub(crate) debug_msg: String,
+}
```

Comment:
This might be one part about error-context I don't understand b/c I think I'm missing where this is used...

Reply:
I don't think it _is_ being used yet; again, I may have copied over more than necessary from the Big PR.